vj4/model/adaptor/contest.py

```diff
@@ -35,6 +35,33 @@ def _oi_stat(tdoc, journal):
   return {'score': sum(d['score'] for d in detail), 'detail': detail}
 
 
+async def _adoj_stat(tdoc, journal):
+  async def apply_penalty(domain_id, pid):
+    p = await document.get(domain_id, document.TYPE_PROBLEM, pid)
+    return p['difficulty'] > 6
+
+  naccept = collections.defaultdict(int)
+  effective = {}
+  for j in journal:
+    if j['pid'] in tdoc['pids'] and not (j['pid'] in effective and effective[j['pid']]['accept']):
+      effective[j['pid']] = j
+      if not j['accept']:
+        naccept[j['pid']] += 1
+  detail = list(dict((j['pid'], j) for j in effective.values() if j['pid'] in tdoc['pids']).values())
+  score = 0
+  for d in detail:
+    if await apply_penalty(tdoc['domain_id'], d['pid']):
+      penalty = naccept[d['pid']]
+    else:
+      penalty = 0
+    realscore = effective.get(d['pid'])['score'] - penalty * 10
```

Comment:
Penalty should be 5 each time.

```suggestion
    realscore = effective.get(d['pid'])['score'] - penalty * 5
```